connectivity.py

```diff
@@ -0,0 +1,100 @@
+from threading import RLock, Event
+
+from nio.properties import TimeDeltaProperty
+from nio.modules.scheduler import Job
+from nio.signal.status import BlockStatusSignal
+from nio.util.runner import RunnerStatus
+
+
+class PubSubConnectivity(object):
+    """ Adds connectivity awareness to pubsub blocks
+    """
+    timeout = TimeDeltaProperty(title='Connect Timeout',
+                                default={'seconds': 2},
+                                advanced=True)
+
+    def __init__(self):
+        super().__init__()
+        self._connected = None
+        self._connected_lock = RLock()
+        self._timeout_job = None
+        self._status_set = {RunnerStatus.warning: False,
+                            RunnerStatus.error: False}
+        self._connected_event = Event()
+
+    def conn_configure(self, is_connected):
+        """ Sets up instance for connectivity checks
+
+        Args:
+            is_connected (callable): function to invoke to establish initial
+                connectivity status
+        """
+        with self._connected_lock:
+            self._connected = is_connected()
+            self.logger.info("Starting in: '{}' state".format(
+                "connected" if self._connected else "disconnected"))
+
+        if not self._connected:
+            # per spec, hold the configure method hoping to get connected
+            if not self._connected_event.wait(self.timeout().total_seconds()):
+                self._notify_disconnection(RunnerStatus.warning)
+                # start changed to error status notification countdown
+                self._timeout_job = Job(self._notify_disconnection,
+                                        self.timeout(), False,
+                                        RunnerStatus.error)
+
+    def conn_stop(self):
+        # cancel existing timeout job if any
+        if self._timeout_job:
+            self._timeout_job.cancel()  # pragma no cover
+
+    def conn_on_connected(self):
+        # remove any possible wait for on_connected event
+        self._connected_event.set()
```

Comment:
I concur.  Since the value of the event and the value of the variable are modified across threads they should both be contained within locks. (due to cpython gil I believe only evaluating the current value of a variable is like an atomic action, but explicit thread safety should be taken)